compiler/src/dmd/escape.d

```diff
@@ -374,24 +374,18 @@ bool checkParamArgumentEscape(Scope* sc, FuncDeclaration fdc, Parameter par, Var
         {
             result |= sc.setUnsafeDIP1000(gag, arg.loc,
                 desc ~ " `%s` assigned to non-scope parameter calling `assert()`", v);
+            return;
         }
-        else if (par)
-        {
-            if (sc.setUnsafeDIP1000(gag, arg.loc,
-                desc ~ " `%s` assigned to non-scope parameter `%s` calling `%s`", v, par, fdc))
-            {
-                result = true;
-                printScopeFailure(previewSupplementalFunc(sc.isDeprecated(), global.params.useDIP1000), vPar, 10);
-            }
-        }
-        else
+        const(char)* msg =
```

Comment:
It's a bit ugly to avoid run-time string concatenation.

Reply:
Is it always evaluated regardless if there's an error?

Reply:
Yes

Reply:
Can this be easily rectified with a delegate? Its better to avoid the whole operation than do a limbo dance avoiding expensive cycles (i.e. allow the code to be simple and avoid the work entirely)

Reply:
Such a delegate would then create a closure on the heap. I wish there was a standard type for error descriptions in DMD, but since we're currently stuck with `printf`-style statements, that's what `setUnsafeDIP1000` is also based on.

Reply:
God forbid a single memory allocation? It's all on the heap already, the GC really struggles with dmd code.

Reply:
In any case, this PR is not about reworking the signature of `setUnsafePreview`.